templates/configure.sh.tpl

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+exec > >(tee /var/log/user-data.log  2>/dev/console) 2>&1
+
+/usr/bin/systemctl stop bitbucket
+
+INSTANCE_ID=`curl -s http://169.254.169.254/latest/meta-data/instance-id`
+
+# wait for ebs volume to be attached
+while true
+do
+    # attach EBS (run multiple times in case the volume was still detaching elsewhere)
+    aws --region ${region} ec2 attach-volume --volume-id ${volume_id} --instance-id $INSTANCE_ID --device /dev/xvdg
+
+    # see if the volume is mounted before proceeding
+    lsblk |grep nvme1n1
+    if [ $? -eq 0 ]
+    then
+        break
+    else
+        sleep 5
+    fi
+done
```

Comment:
Doing this in a while loop, even with a 5 second sleep, is a bad idea because of throttling. It's not terribly difficult to implement exponential backoff.

https://docs.aws.amazon.com/general/latest/gr/api-retries.html

Reply:
Also you can loop on checking the status (to wait for it to be available) rather than just blindly trying to attach

Reply:
I'm not sure I see a reason to make this change. It's a fairly simple, very common design pattern and is massively higher than the throttling limits that AWS has (which are measured in ms, not s).